reconcile/alertmanager_config_validator.py

```diff
@@ -0,0 +1,88 @@
+import sys
+import semver
+import base64
+import logging
+
+import reconcile.utils.gql as gql
+import reconcile.utils.threaded as threaded
+import reconcile.utils.amtool as amtool
+import reconcile.openshift_resources_base as orb
+
+from reconcile.status import ExitCodes
+
+QONTRACT_INTEGRATION = 'alertmanager_config_validator'
+QONTRACT_INTEGRATION_VERSION = semver.format_version(0, 1, 0)
+
+ALERTMANAGER_SECRET_PATH = '/observability/alertmanager/' \
```

Comment:
would it make sense to mark the resources to check in the schema instead? for grafana dashboards we have `validate_json: true`.

Reply:
Hmmm, are you proposing that we add a new property, e.g. `validate_alertmanager_config` to trigger this validation instead of the horrible path hardcoding? Because I like that approach much more than the current one 😂

Reply:
yes! perhaps even `validate_alertmanager_config_key: alertmanager.yaml` to catch 2 birds with 1 stone?

Reply:
While the idea of catching two birds with one stone is tempting, I see this more properly done with two properties:

`validate_alertmanager_config`: <boolean>
`validate_alertmanager_config_key`: string with a default of `alertmanager.yaml`

Reply:
leaving the implementation decision to you